core/src/main/java/org/apache/airavata/mft/core/api/ConnectorConfig.java

```diff
@@ -90,6 +90,40 @@ public void setTransportConfig(Map transportConfig) {
         this.transportConfig = transportConfig;
     }
 
+    public static final class LocalConfigs {
+        public LocalConfigs() {
+        }
+
+        public final static String DMA_ENABLED = "local.dma";
+        public final static String BUFF_LEN = "local.buffLen";
+    }
+
+    public Boolean getBooleanTransportProperty(String key, Boolean defaultValue){
+        if(key == LocalConfigs.DMA_ENABLED){
+            if (!this.transportConfig.get(key).toString().isEmpty()) return Boolean.getBoolean(this.transportConfig.get(key).toString());
```

Comment:
This logic throws an error if the key is not in the map. I would suggest the following logic

`return this.transportConfig.containesKey(key)? Boolean.parseBoolean(this.transportConfig.get(key)): defaultValue`

Reply:
okay